test/cfg/qt.cpp

```diff
@@ -30,6 +31,44 @@ int QString2()
     return s.size();
 }
 
+void QVector1(QVector<int> intVectorArg)
+{
+    for (int i = 0; i <= intVectorArg.size(); ++i) {
+        // cppcheck-suppress stlOutOfBounds
+        intVectorArg[i] = 1;
+    }
+    // cppcheck-suppress containerOutOfBoundsIndexExpression
+    intVectorArg[intVectorArg.length()] = 5;
+    // cppcheck-suppress containerOutOfBoundsIndexExpression
+    intVectorArg[intVectorArg.count()] = 10;
+    // cppcheck-suppress containerOutOfBoundsIndexExpression
+    printf("val: %d\n", intVectorArg[intVectorArg.size()]);
+
+    QVector<QString> qstringVector1{"one", "two"};
+    (void)qstringVector1[1];
+
+    QVector<QString> qstringVector2 = {"one", "two"};
+    (void)qstringVector2[1];
+
+    QVector<QString> qstringVector3;
+    qstringVector3 << "one" << "two";
+    //(void)qstringVector3[1]; // TODO: no containerOutOfBounds error should be shown
```

Comment:
@danmar Is there a way to tell Cppcheck that `<<` operator also "pushes" values and changes the size?
I tried this, but that does not work (because `<<` characters are not allowed here I guess):
```xml
    <size>
      <function name="&lt;&lt;" action="push"/>
    </size>
```
There should be `std` containers where it is similar (for example `std::string`), but i can not see where this is configured there.

Reply:
No... it sounds good to add some such functionality.

Reply:
Ok, thanks